packages/python/plotly/plotly/express/_core.py

```diff
@@ -1419,9 +1419,17 @@ def build_dataframe(args, constructor):
             else:
                 # Save precious resources by only interchanging columns that are
                 # actually going to be plotted.
-                columns = [
+                necessary_columns = [
                     i for i in args.values() if isinstance(i, str) and i in columns
                 ]
+                for field in args:
+                    if field in array_attrables and isinstance(
+                        args[field], (list, dict)
+                    ):
```

Comment:
`px` is generally much looser with types, and elsewhere seems to assume `array_attrables` are iterables already. So can we remove this `isinstance` check so that other iterables will work too? I don't think we even need to exclude `str`, as a string would already have caused other problems for these args. I'm thinking of tuples at least but I'm sure there are others that will arise. Another example, `reversed([1,2,3])` is not a list...

Reply:
You're right!
I used this condition to ignore string arguments, but if you say they are handled elsewhere then sure, why not simplify this code :)

Reply:
You're right!
I used this condition to ignore string arguments, but if you say they are handled elsewhere then sure, why not simplify this code :)